chunked processing fixes and reduce v1f syscalls

nigoroll opened the pull request:
* v1f performance:
v1f called reads even on one-byte buffers. As we normally have a bit of pipeline buffer space from session init, we can re-use it for some readahead. As the readahead must not block, we need to switch blocking on and off more frequently, so we move blocking to the HTC layer and cache the current blocking state.
* chunked:
  * fix parsing of chunk extensions (as required by httpbis)
  * fix handing of trailers (by ignoring them for now, but I want to add full trailer support)

I have *intentionally* left in two disabled Debug VSLs to make life easier for reviewers.

nigoroll commented:
FTR I've force-pushed a minor fix to the recursion in `v1f_read()` and stabilized the test. The test failure previously reported by Travis was caused by the fact that I had several sends for the trailers so vtc could complain about the close from varnishd (timing sensitive)
the current code survives `./varnishtest -n 200 -j 40 -t 10 -i tests/b00007.vtc` on my box

bsdphk commented:
I don't understand the need for the rxra fields, how are they different from the pipeline fields ?

nigoroll commented:
@bsdphk once the pipeline is consumed, the pipeline fields are nulled. the rxra fields just point to the pipeline region for re-use

nigoroll commented:
rebased and added dae7d5aeaedb345a8e8db8e104d93f3c4de370f3 - this would be squeezed with the previous commit for merge, but keeping it separate for reviewers

nigoroll commented:
continued in #2477 ...
